src/libraries/System.Reflection.Metadata/src/System/Reflection/Metadata/BlobReader.cs

```diff
@@ -437,6 +438,16 @@ public void ReadBytes(int byteCount, byte[] buffer, int bufferOffset)
             Marshal.Copy((IntPtr)GetCurrentPointerAndAdvance(byteCount), buffer, bufferOffset, byteCount);
         }
 
+        string IBinaryReader.ReadNullPaddedUTF8(int byteCount)
+        {
+            byte* ptr = GetCurrentPointerAndAdvance(byteCount);
+            while (byteCount > 0 && ptr[byteCount - 1] == 0)
+            {
+                byteCount--;
+            }
```

Comment:
This function gets called only once with a constant size of 8. Providing a separate implementation with spans for modern .NET is not worth it.